src/mdlComponentHandler.js

```diff
@@ -390,7 +366,16 @@ componentHandler = (function() {
      * @param  {!Node} node the node to be downgraded
      */
     var downgradeNode = function(node) {
-      deconstructComponentInternal(findCreatedComponentByNodeInternal(node));
+      var found = [];
+      for (var n = 0; n < createdComponents_.length; n++) {
```

Comment:
Pushing the filter/forEach method. It is a hell of a lot cleaner and I really don't think we need to hold back the code style based on consistency with outdated techniques.